pkg/collector/metric/process_metric.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metric
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/sustainable-computing-io/kepler/pkg/bpfassets/attacher"
+	"github.com/sustainable-computing-io/kepler/pkg/config"
+	"github.com/sustainable-computing-io/kepler/pkg/power/accelerator"
+	"k8s.io/klog/v2"
+)
+
+var (
+	// ProcessMetricNames holds the list of names of the container metric
+	ProcessMetricNames []string
+	// ProcessFloatFeatureNames holds the feature name of the container float collector_metric. This is specific for the machine-learning based models.
+	ProcessFloatFeatureNames []string = []string{}
+	// ProcessUintFeaturesNames holds the feature name of the container utint collector_metric. This is specific for the machine-learning based models.
+	ProcessUintFeaturesNames []string
+	// ProcessFeaturesNames holds all the feature name of the container collector_metric. This is specific for the machine-learning based models.
+	ProcessFeaturesNames []string
+)
+
+type ProcessMetrics struct {
+	PID          uint64
+	Command      string
+	CounterStats map[string]*UInt64Stat
+	// ebpf metrics
+	CPUTime           *UInt64Stat
+	SoftIQRCount      []UInt64Stat
```

Comment:
Shouldn't this be SoftIRQ instead of SoftIQR

Reply:
nice catch!